packages/salesforcedx-vscode-core/src/orgPicker/orgList.ts

```diff
@@ -88,34 +91,26 @@ export class OrgList implements vscode.Disposable {
         // scratch orgs parented by other (non-default) devHub orgs
         continue;
       }
-      const isExpired =
-        authFields && authFields.expirationDate
-          ? today >= new Date(authFields.expirationDate)
-          : false;
 
+      if (orgAuth.isExpired === true) {
+        // If the scratch org is expired we don't want to see it in the org picker
+        continue;
+      }
       const aliases = await ConfigUtil.getAllAliasesFor(orgAuth.username);
```

Comment:
I was digging around in the OrgAuthorization and it also has aliases, so maybe we don't have to repeat the fetch here.

See https://github.com/forcedotcom/sfdx-core/blob/da5483e7bbbdd1837f313d168fe5aeaa5e5ec016/src/org/authInfo.ts#L90